mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java

```diff
@@ -100,8 +106,10 @@ class ResourceClusterActor extends AbstractActorWithTimers {
 
     private final ResourceClusterActorMetrics metrics;
 
-    static Props props(final ClusterID clusterID, final Duration heartbeatTimeout, Duration assignmentTimeout, Duration disabledTaskExecutorsCheckInterval, Clock clock, RpcService rpcService, MantisJobStore mantisJobStore, JobMessageRouter jobMessageRouter) {
-        return Props.create(ResourceClusterActor.class, clusterID, heartbeatTimeout, assignmentTimeout, disabledTaskExecutorsCheckInterval, clock, rpcService, mantisJobStore, jobMessageRouter);
+    private final ConcurrentMap<ArtifactID, ArtifactID> jobArtifactsToCache = new ConcurrentHashMap<>();
```

Comment:
nit: This can be converted to a HashMap I'm guessing after your current set of changes.